server/adaptors/opensearch_observability_plugin.ts

```diff
@@ -51,6 +51,19 @@ export function OpenSearchObservabilityPlugin(Client: any, config: any, componen
     method: 'GET',
   });
 
+  observability.getDatasourceById = clientAction({
+    url: {
+      fmt: `${OPENSEARCH_DATASOURCES_API.DATASOURCE}/<%=datasource%>`,
+      req: {
+        datasource: {
+          type: 'string',
+          required: true,
+        },
+      },
+    },
+    method: 'GET',
+  });
+
```

Comment:
If data sources a new API in observablity backend? If not then let's decouple them.

Reply:
It is an API in sql backend - 
OPENSEARCH_DATASOURCES_API is pointed to `/_plugins/_query/_datasources`

Reply:
I am using PPLService to run `show datasources` based on here: https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/showdatasources.rst#example-1-fetch-all-prometheus-datasources for the other query. This is to get the info for a specific given datasource

Reply:
This makes sense, thanks!

Reply:
Moved to PPL route - since that file is all relying on sql backend